### DISCUSSION TITLE
Is there any way to bypass the download size limit of 20MB ?

### DISCUSSION THREAD

**Discussion by XDavidT** [aiogram/aiogram#557] [Q&A] [answered]
As my boot need to download file, proccess it, and return to user, I found that telegram is blocking me if the file is too big.
Found some answers in StackOverflow say you can request the file directly from the CDN.
Is there any option do so with aiogram ? Or maybe use this bypass and manipulate ?

**Comment by JrooTJunior** [comment, marked as answer]
You can't do this via public Bot API in due to their limitations: https://core.telegram.org/bots/api#getfile, https://core.telegram.org/bots/api#sending-files

But you can do this via local Bot API Server: https://core.telegram.org/bots/api#using-a-local-bot-api-server
Or just use Client API directly for working with files. local Bot API Server has the same limitations as Client API.

### Local server:

Can be compiled from [official repository](https://github.com/tdlib/telegram-bot-api) or used via third-party Docker images https://hub.docker.com/r/aiogram/telegram-bot-api
And `--local` mode is required (`TELEGRAM_LOCAL=1` env. var if you use my Docker image)

### Client API:

*aiogram* can be used at the same time with one of Client API library like [Pyrogram](https://docs.pyrogram.org/), [Telethon](https://docs.telethon.dev/en/latest/) or [tdlib](https://github.com/tdlib/td) for working with files.

**Reply by Dadamuhames** [reply]
> Client API:
aiogram can be used at the same time with one of Client API library like [Pyrogram](https://docs.pyrogram.org/), [Telethon](https://docs.telethon.dev/en/latest/) or [tdlib](https://github.com/tdlib/td) for working with files.


How?

**Reply by JrooTJunior** [reply]
Just read the docs for this libraries.
